build-presets/MakePresetTree.py

```diff
@@ -1,86 +1,127 @@
-            'cost_back_layer': module_details['cost_back_layer'][package_type],
-            'cost_noncell': module_details['cost_noncell'],
-            'cost_om': cost_om[system_type],
-            'efficiency': module_details['efficiency'][cell_technology],
-            'energy_yield': np.round(system_yield[(cell_technology, package_type, system_type, location)]),
-            'degradation_rate': degradation_rates[(cell_technology, package_type, system_type, location)],
-            'state': location.split(' ')[1]
-        }
-
-
-# %% Save as javascript files
-with open('../js/preset_tree.min.js', 'w') as file:
-    file.write('var preset_tree = ' + json.dumps(preset_tree, separators=(',', ':')))
-# Create preset_tree.js if you want a version that is more legible to humans    
-with open('../js/preset_tree.js', 'w') as file:
+for cell_technology in cell_technologies:
+	if cell_technology not in preset_tree:
+		preset_tree[cell_technology] = {}
+	for package_type in package_types[cell_technology]:
+		if package_type not in preset_tree[cell_technology]:
+			preset_tree[cell_technology][package_type] = {}
+		for system_type in system_types[cell_technology]:
+			if system_type not in preset_tree[cell_technology][package_type]:
+				preset_tree[cell_technology][package_type][system_type] = {}
+			for ilr in inverter_loading_ratio:
+				if ilr not in preset_tree[cell_technology][package_type][system_type]:
+					preset_tree[cell_technology][package_type][system_type][ilr] = {}
+				for solar_resource_file in weather_data:
+					
+					json_model.SolarResource.solar_resource_data = solar_resource_file
+					lat = json_model.SolarResource.solar_resource_data['lat']
+					lon = json_model.SolarResource.solar_resource_data['lon']
+					location = locations[(lat, lon)] # string name of location
+
+					# set specific inputs of PySAM model based on system type and ILR
+					json_model.SystemDesign.gcr = ground_coverage_ratio[system_type]
+					json_model.SystemDesign.array_type = array_type[system_type]
+					json_model.SystemDesign.tilt = tilt[system_type]
+					json_model.SystemDesign.dc_ac_ratio = ilr
+
+					json_model.execute(0) # run the model
+					energy_yield = json_model.Outputs.kwh_per_kw # get the energy yield from the outputs
+
+					preset_tree[cell_technology][package_type][system_type][ilr][location] = {
+						'cost_front_layer': module_details['cost_front_layer'],
+            					'cost_cell': module_details['cost_cell'][cell_technology],
+           					'cost_back_layer': module_details['cost_back_layer'][package_type],
+            					'cost_noncell': module_details['cost_noncell'],
+            					'cost_om': cost_om[system_type],
+            					'efficiency': module_details['efficiency'][cell_technology],
+            					'energy_yield': energy_yield,
+            					'degradation_rate': 0.7,
+            					'state': location.split(' ')[1]
+					}
+
+# print(preset_tree)			
+with open('../js/PresetTree.js', 'w') as file:
```

Comment:
This may not be platform independent. Along with `from pathlib import Path` up top, my untested suggestion is:
```suggestion
with open(Path('../js/PresetTree.js'), 'w') as file:
```